libs/langchain-google-webauth/src/tests/chat_models.int.test.ts

```diff
@@ -0,0 +1,247 @@
+        resArray.push(chunk);
+      }
+      expect(resArray).toBeDefined();
+      expect(resArray.length).toBeGreaterThanOrEqual(1);
+
+      const lastChunk = resArray[resArray.length - 1];
+      expect(lastChunk).toBeDefined();
+      expect(lastChunk._getType()).toEqual("ai");
+      const aiChunk = lastChunk as AIMessageChunk;
+      console.log(aiChunk);
+
+      console.log(JSON.stringify(resArray, null, 2));
+    } catch (e) {
+      console.error(e);
+      throw e;
+    }
+  });
+
+  test.skip("Tool call", async () => {
+    const chat = new ChatGoogle().bindTools([new WeatherTool()]);
+    const res = await chat.invoke("What is the weather in SF and LA");
+    console.log(res);
+    expect(res.tool_calls?.length).toEqual(1);
+    expect(res.tool_calls?.[0].args).toEqual(
+      JSON.parse(res.additional_kwargs.tool_calls?.[0].function.arguments ?? "")
+    );
+  });
+
+  test.skip("Few shotting with tool calls", async () => {
+    const chat = new ChatGoogle().bindTools([new WeatherTool()]);
+    const res = await chat.invoke("What is the weather in SF");
+    console.log(res);
+    const res2 = await chat.invoke([
+      new HumanMessage("What is the weather in SF?"),
+      new AIMessage({
+        content: "",
+        tool_calls: [
+          {
+            id: "12345",
+            name: "get_current_weather",
+            args: {
+              location: "SF",
+            },
+          },
+        ],
+      }),
+      new ToolMessage({
+        tool_call_id: "12345",
+        content: "It is currently 24 degrees with hail in SF.",
+      }),
+      new AIMessage("It is currently 24 degrees in SF with hail in SF."),
+      new HumanMessage("What did you say the weather was?"),
+    ]);
+    console.log(res2);
+    expect(res2.content).toContain("24");
+  });
+
+  test.skip("withStructuredOutput", async () => {
+    const tool = {
+      name: "get_weather",
+      description:
+        "Get the weather of a specific location and return the temperature in Celsius.",
+      parameters: {
+        type: "object",
+        properties: {
+          location: {
+            type: "string",
+            description: "The name of city to get the weather for.",
+          },
+        },
+        required: ["location"],
+      },
+    };
+    const model = new ChatGoogle().withStructuredOutput(tool);
+    const result = await model.invoke("What is the weather in Paris?");
+    expect(result).toHaveProperty("location");
+  });
+
+  test("media - fileData", async () => {
+    const aliasStore = new BackedBlobStore({
+      backingStore: new InMemoryStore()
+    });
+    const canonicalStore = new BlobStoreAIStudioFile({
+    })
+    const resolver = new SimpleWebBlobStore();
+    const mediaManager = new MediaManager({
+      aliasStore,
+      canonicalStore,
+      resolver,
+    })
+    const model = new ChatGoogle({
+      modelName: "gemini-1.5-flash",
+      apiVersion: "v1beta",
+      mediaManager,
```

Comment:
I think we have to add this to the chat model class above?

Also - thoughts on calling it `googleMediaManager` for now?